File: 5.py
class Solution(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: str
        """
        a=['#']
        for i in s:
            a.append(i)
            a.append('#')
        
        p=0
        ll=0
        for i in range(len(a)):
            l=0
            while i-l-1>=0 and i+l+1<len(a) and a[i-l-1]==a[i+l+1]:
                l=l+1
                if l>ll:
                    ll=l
                    p=i
        
        c=[]
        for i in range(p-ll,p+ll):
            if a[i]!='#':
                c.append(a[i])
        d="".join(c)
        return d

File: algorithms/5.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        
        string a;
        a.push_back('#');
        for(auto i:s){
            a.push_back(i);
            a.push_back('#');
        }
        int l=0;
        int p=0;
        int ll=0;
        for(int i=0;i<a.size();i++){
            l=0;
            while(i-l-1>=0&&i+l+1<a.size()&&a[i-l-1]==a[i+l+1]){
                l++;
                if(l>ll){
                    //cout<<"!"<<a[i-l-1]<<" "<<a[i+l+1]<<" "<<i<<l<<endl;
                    ll=l;
                    p=i;
                }
            }
        }
        //cout<<a<<endl;
        //cout<<p<<endl;
        //cout<<ll<<endl;
        string c;
        for(int i=p-ll;i<=p+ll;i++){
            if(a[i]!='#') c.push_back(a[i]);
        }
        
        
        return c;
        
    }
};
